packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/FunctionsTests.kt

```diff
@@ -0,0 +1,623 @@
+//    }
+//
+//    @Test
+//    fun asyncCodecBsonEncodingFailure() = looperThread.runBlocking {
+//        functions.callFunctionAsync(FIRST_ARG_FUNCTION, listOf(32), String::class.java, faultyIntegerRegistry) { result ->
+//            try {
+//                assertEquals(ErrorCode.BSON_ENCODING, result.error.errorCode)
+//            } finally {
+//                looperThread.testComplete()
+//            }
+//        }
+//    }
+//
+//    @Test
+//    fun codecBsonDecodingFailure() {
+//        assertFailsWithErrorCode(ErrorCode.BSON_DECODING) {
+//            functions.callFunction(FIRST_ARG_FUNCTION, listOf(32), String::class.java)
+//        }
+//    }
+//
+//    @Test
+//    fun asyncCodecBsonDecodingFailure() = looperThread.runBlocking {
+//        functions.callFunctionAsync(FIRST_ARG_FUNCTION, listOf(32), String::class.java) { result ->
+//            try {
+//                assertEquals(ErrorCode.BSON_DECODING, result.error.errorCode)
+//                assertTrue(result.error.exception is BSONException)
+//            } finally {
+//                looperThread.testComplete()
+//            }
+//        }
+//    }
+//
+//    @Test
+//    fun localCodecRegistry() {
+//        val input = Dog("PojoFido")
+//        assertEquals(input, functions.callFunction(FIRST_ARG_FUNCTION, listOf(input), Dog::class.java, pojoRegistry))
+//    }
+//
+//    @Test
+//    fun asyncLocalCodecRegistry() = looperThread.runBlocking {
+//        val input = Dog("PojoFido")
+//        functions.callFunctionAsync(FIRST_ARG_FUNCTION, listOf(input), Dog::class.java, pojoRegistry) { result ->
+//            try {
+//                assertEquals(input, result.orThrow)
+//            } finally {
+//                looperThread.testComplete()
+//            }
+//        }
+//    }
+//
+//    @Test
+//    fun instanceCodecRegistry() {
+//        val input = Dog("PojoFido")
+//        val functionsWithCodecRegistry = anonUser.getFunctions(pojoRegistry)
+//        assertEquals(input, functionsWithCodecRegistry.callFunction(FIRST_ARG_FUNCTION, listOf(input), Dog::class.java))
+//    }
+//
+//    @Test
+//    fun resultDecoder() {
+//        val input = "Realm"
+//        val output = "Custom Realm"
+//        assertEquals(output, functions.callFunction(FIRST_ARG_FUNCTION, listOf(input), CustomStringDecoder(output)))
+//    }
+//
+//    @Test
+//    fun asyncResultDecoder() = looperThread.runBlocking {
+//        val input = "Realm"
+//        val output = "Custom Realm"
+//        functions.callFunctionAsync(FIRST_ARG_FUNCTION, listOf(input), CustomStringDecoder(output), App.Callback<String> { result ->
+//            try {
+//                assertEquals(output, result.orThrow)
+//            } finally {
+//                looperThread.testComplete()
+//            }
+//        })
+//    }
+
+    @Test
+    fun unknownFunction() {
+        assertFailsWithMessage<FunctionExecutionException>("[Service][FunctionNotFound(26)] function not found: 'unknown'") {
```

Comment:
Maybe just assert on the relevant explanatory part of the message "function not found"